src/generators.rs

```diff
@@ -551,6 +551,329 @@ pub fn star_graph(
+                let index = graph.add_node(weight);
+                node_list.push(index);
+                node_cnt -= 1;
+            }
+            for _i in 0..node_cnt {
+                let index = graph.add_node(py.None());
+                node_list.push(index);
+            }
+            node_list
+        }
+        None => (0..num_nodes).map(|_| graph.add_node(py.None())).collect(),
+    };
+
+    for i in 0..rowlen {
+        for j in 0..collen {
+            if i + 1 < rowlen {
+                graph.add_edge(
+                    nodes[i * collen + j],
+                    nodes[(i + 1) * collen + j],
+                    py.None(),
+                );
+            }
+            if j + 1 < collen {
+                graph.add_edge(
+                    nodes[i * collen + j],
+                    nodes[i * collen + j + 1],
+                    py.None(),
+                );
+            }
+        }
+    }
+    Ok(graph::PyGraph {
+        graph,
+        node_removed: false,
+    })
+}
+
+/// Generate a directed grid graph. The edges propagate towards right and
+///     bottom direction if ``bidirectional`` is ``false``
+///
+//// :param int rows: The number of rows to generate the graph with.
+///     If specified, cols also need to be specified.
+/// :param list cols: The number of rows to generate the graph with.
+///     If specified, rows also need to be specified. rows*cols
+///     defines the number of nodes in the graph.
+/// :param list weights: A list of node weights. If rows and cols
+///     are not specified, then a linear graph containing all the
+///     values in weights list is created.
+///     If number of nodes(rows*cols) is less than length of
+///     weights list, the trailing weights are ignored.
+///     If number of nodes(rows*cols) is greater than length of
+///     weights list, extra nodes with None weight are appended.
+/// :param bidirectional: A parameter to indicate if edges should exist in
+///     both directions between nodes
+///
+/// :returns: The generated star graph
+/// :rtype: PyDiGraph
+/// :raises IndexError: If neither ``rows`` or ``cols`` and ``weights`` are
+///      specified
+///
+#[pyfunction(bidirectional = "false")]
+#[text_signature = "(/, rows=None, cols=None, weights=None, bidirectional=False)"]
+pub fn directed_grid_graph(
+    py: Python,
+    rows: Option<usize>,
+    cols: Option<usize>,
+    weights: Option<Vec<PyObject>>,
+    bidirectional: bool,
+) -> PyResult<digraph::PyDiGraph> {
+    let mut graph = StableDiGraph::<PyObject, PyObject>::default();
+    if weights.is_none() && (rows.is_none() || cols.is_none()) {
+        return Err(PyIndexError::new_err(
+            "dimensions and weights list not specified",
+        ));
+    }
+
+    let mut rowlen = match rows {
+        Some(rows) => rows,
+        None => 0,
+    };
+
+    let mut collen = match cols {
+        Some(cols) => cols,
+        None => 0,
+    };
+
+    let mut num_nodes = rowlen * collen;
```

Comment:
What happens if num_nodes is 0? Also what happens if weights.len() != num_nodes? I think we should probably add error handling to cover the case where someone calls: `grid_graph(0, 0)` or `grid_graph(100, 100, ['a', 'b'])`

Reply:
I have handled the case grid_graph(100, 100, ['a', 'b']) by appending Py.None() weights. So in this case 100*100 - 2 nodes are added into the graph. For grid_graph(0, 0) an empty graph is returned. Do we need to error out both the cases?

Reply:
Nah, I think it's fine like this as long as we document this behavior and assert it with tests.

Reply:
I had added the documentation and tests. Let me know if they need to be updated

Reply:
Thanks, yeah it looks fine now.